helper/staking/staking.go

```diff
@@ -134,12 +156,22 @@ func PredeployStakingSC(
 		return nil, fmt.Errorf("unable to generate DefaultStatkedBalance, %w", err)
 	}
 
+	//Validate min and max validators number
+	if err = validateMinMaxValidatorsNumber(params.MinValidatorCount, params.MaxValidatorCount); err != nil {
+		return nil, err
+	}
+
 	// Generate the empty account storage map
 	storageMap := make(map[types.Hash]types.Hash)
 	bigTrueValue := big.NewInt(1)
 	stakedAmount := big.NewInt(0)
+	minNumValidators := big.NewInt(int64(params.MinValidatorCount))
+	maxNumValidators := big.NewInt(int64(params.MaxValidatorCount))
 
 	for indx, validator := range validators {
+		if int64(indx) == maxNumValidators.Int64() {
+			break
+		}
```

Comment:
This check shouldn't even exist here.

Why are you breaking if there is a mismatch between the number of validators in the validator set and the max count? Why even have this condition here, you should be certain when going into `PredeployStakingSC` that `len(validators) == maxNumValidators`.

Please remove this check, and make sure that the validator set you pass into the method is matches the limits size-wise

Reply:
fixed a91c866e0f6d742a0cc4af63f07196b4703ec96a

@Kourin1996 suggested that we should return error if len(validators) > maxNumValidators because user is giving them as validators

Reply:
Refactor 081af15b3e97e5e7ea3c38ee6b9c619af6ce2673